### DISCUSSION TITLE
Column Order

### DISCUSSION THREAD

**Discussion by mvit777** [EvotecIT/PSWriteHTML#371] [General]
Probably asked before:
Column order seems to be random (does not seem to be alphabetical or other)
Is there a parameter for column order?

p.s.
I'm using Out-HtmlView rather than DataTable

**Comment by PrzemyslawKlys** [comment]
Column order should be preserved unless you are using hashtable?

**Comment by mvit777** [comment]
I'm using an array objects read from a json file. Is that the case?

**Reply by PrzemyslawKlys** [reply]
It should be PSCustomObject when you do that. So order should be retained. Aer you using any specific parameters?

**Comment by mvit777** [comment]
$config.terms below I confirm is an array of PSCustomObject and I use Newtonsoft.json to deserialize.
Parameters are as follows
`
$config.terms |Out-HtmlView -InvokeHTMLTags -PagingLength 30 -FilePath $PSScriptRoot\Glossary.html
`
In the meanwhile thank you for the beatiful library

**Reply by PrzemyslawKlys** [reply]
So the order should be correct, but it will take the order from the first object in the array.  Can you provide minimal example that I can check?

**Reply by mvit777** [reply]
Well I tried to print the $config.terms and with my surprise the props in the object array are ordered randomly.
Also tried using Out-GridView in place of Out-HtmlView and it popped up the same way.
I cannot figure out where the issue stems from but clearly it has  nothing to do with your library.
Sorry for wasting your time and thanks indeed for the help and the excellent lib.

**Reply by PrzemyslawKlys** [reply]
You can always use ResponsivePrioriotyOrder or/and PriorityProperties or IncludeProperty to force the right order. You can also use Seelect-Object before you pass it to Out-Htmlview.

**Comment by mvit777** [comment]
PriorityProperties did the trick!
Thanks a lot
